device-server/src/main/kotlin/com/badoo/automation/deviceserver/ios/simulator/Simulator.kt

```diff
@@ -637,4 +643,28 @@ class Simulator (
         logger.debug(logMarker, "Uninstalling application $bundleId from Simulator $this")
         remote.execIgnoringErrors(listOf("xcrun", "simctl", "uninstall", udid, bundleId))
     }
+
+    override fun setEnvironmentVariables(envs: Map<String, String>) {
+        logger.debug(logMarker, "Setting environment variables $envs for Simulator $this")
+        envs.keys.forEach {
+            remote.shell("xcrun simctl spawn $udid launchctl setenv $it ${envs[it]}")
```

Comment:
I am concerned that envs[it] are not escaped, which will potentially cause a problem if I want to set env variable to a value containing space. For example `TEST=123 123`.
I think you can use `ShellUtils.escape`, but needs to be tested for local and ssh modes.

Also, it looks like `launchctl setenv` supports setting multiple env variables in a single call, this will save a lot on process calls, especially over ssh. See:
```
Usage: launchctl setenv <<key> <value>, ...>
```

Reply:
Done.
#34

I've tried different variants with `execIgnoringErrors()`, but they didn't work locally. Method with `remote.shell()` and `ShellUtils.escape()` on variable value works fine.